packages/web-lib/types/index.ts

```diff
@@ -17,9 +17,4 @@ export type NDict<T> = TNDict<T>;
 export type VoidPromiseFunction = () => Promise<void>;
 
 // TAddress is used to represent a checksummed address
-export type	TAddress = '/^0x([0-9a-f][0-9a-f])*$/I'
-
-// TMetamaskInjectedProvider is used to represent a Metamask injected provider
-export type TMetamaskInjectedProvider = ethers.providers.BaseProvider & {
-	send: (...args: any[]) => void;
-}
+export type	TAddress = `0x${string}`
```

Comment:
This is not good, we should definitely keep the regular expression for the `TAddress` type, we can use the one I'm working on here https://github.com/yearn/yearn.fi/pull/165/files#diff-0cd290cb864ef7134af8b8cdb5dd9448795beeccf88fcec650ac7115ca1579a7

Now `"0xfoobar"` is a valid `TAddress`

Reply:
I think what you currently have is the way, there's really not much we can do. The `isAddress` guard works fine to guarantee that we're not getting "0xFoo" as an address